src/main/java/io/debezium/connector/vitess/connection/VitessReplicationConnection.java

```diff
@@ -194,6 +194,7 @@ private int getNumOfRowEvents(Vtgate.VStreamResponse response) {
     private ManagedChannel newChannel(String vtgateHost, int vtgatePort) {
         ManagedChannel channel = ManagedChannelBuilder.forAddress(vtgateHost, vtgatePort)
                 .usePlaintext()
+                .keepAliveTime(config.getKeepaliveInterval().toMillis(), TimeUnit.MILLISECONDS)
```

Comment:
It is not super clear to me why the channel get closed before this change - for my own understanding, is the channel closed by grpc because it is idle? I think the default idle timeout was 30 min: https://grpc.github.io/grpc-java/javadoc/io/grpc/ManagedChannelBuilder.html#idleTimeout-long-java.util.concurrent.TimeUnit-

Reply:
After troubleshooting for quite a bit I couldn't make a conclusion. I tried to set idleTimeout to different values and there was no effect at all. I added debug output and found the channel itself is mostly in IDLE state even before the timeout, so this idle setting might be irrelevant. Because there is only one gRPC activity (VStream) and no new gRPCs. I also debugged server-side as well. 

The general assumption is that the connection/transport is closed after no active streams for a while. To gRPC this means no pings between peers for a period, so the transport is disconnected. I have searched and found these references:
* https://stackoverflow.com/questions/57930529/grpc-connection-use-keepalive-or-idletimeout
* https://stackoverflow.com/questions/64119639/when-should-a-channel-be-closed-on-grpc
* https://grpc.github.io/grpc/core/md_doc_connectivity-semantics-and-api.html
* https://confluence.corp.stripe.com/display/SN/Delayed+Connection+Draining
https://github.com/grpc/grpc/blob/master/doc/keepalive.md
https://cs.mcgill.ca/~mxia3/2019/02/23/Using-gRPC-in-Production/

Reply:
The links seem to suggest using idealTimeout over keepAliveTime 🤷  

I think it's fine to add this option here, the only concern I have here is it requires corresponding setting on the server side as well (e.g., [grpc.keepalive_permit_without_calls](https://grpc.github.io/grpc/core/group__grpc__arg__keys.html#gaf900669f52f137677c4dbb9a7a902c92))

I'm not sure in this case if it worth adding a TODO for it, I will defer to you and others here

Reply:
yeah, idealTimeout doesn't work in my verification so concluded that it's not related to idleTimeout, because idleTimeout is about "channel", but our problem is with "transport/connection", which we need to tackle with keepalive pings.

As for keepalive_permit_without_calls, I tested without this option and it works. According to [this](https://github.com/grpc/grpc/blob/master/doc/keepalive.md) and [this](https://grpc.github.io/grpc-java/javadoc/io/grpc/ManagedChannelBuilder.html#keepAliveWithoutCalls-boolean-):
* allows keepalive pings to be sent even if there are no calls in flight
* Sets whether keepalive will be performed when there are no outstanding RPC on a connection

I think we do have a "call" or outstanding RPC, that is, the VStream call itself. Therefore without this option it won't affect our keepalive pings. I will need to test it in prod for several passes once I had the snapshots ready. We can add it if we do need this option.